### ISSUE TITLE
Always shows Czech translation even on `/en/` version of the site

### ISSUE THREAD

**Issue by matronator** [contributte/translation#100]
Everything was working as it should, the default locale was `en` and you could change to `cs` or `ru` by redirecting to `site.com/en` or `site.com/ru`.

But today I needed to update some Nette packages and I foolishly ran `composer update` which updated everything including some translation packages:

```yaml
  - Upgrading symfony/translation-contracts (v2.3.0 => v2.5.1): Extracting archive
  - Upgrading symfony/translation (v5.2.3 => v5.4.9): Extracting archive
  - ...
```

And now everything is in Czech, no matter what language I select. Also after checking Tracy, it seems there is some `Header` listed in `Locale resolvers` which is always set to `cs` and is listed last, so I assume it overrides all the resolvers before. What is this Header resolver and how do I get rid of it, so the locales are changed based on the Router parameter again?

Tracy screenshot:

<img width="1013" alt="Snímek obrazovky 2022-06-01 v 21 38 53" src="https://user-images.githubusercontent.com/5470780/171489377-04fe07e0-6a22-45fb-ad59-ed29f0c67287.png">

---

**EDIT:** Okay, after further digging, it seems that on every request there is this header `Accept-Language: cs-cz`, which might be the cause of the problem. Why it didn't matter before is beyond me - was it even sent before the update?

Anyways, how do I get rid of either the header itself or (preferably) the Header resolver or at least lower its priority below Route parameter?

**Comment by aleswita** [comment]
If you rollback composer update, solved the problem?

**Comment by matronator** [comment]
I tried to do that, but I was messing it up somehow, because it always installed the new versions of everything even after deleting `composer.lock` and `vendor/`. I only just now thought about reverting `composer.lock` to what it was before, but it's too late for it now as I have already made many changes to accommodate the new versions of everything (where things broke... It was the first update after more than a year) so that would also have to be reverted along with all the new features.

But I found a "hacky" solution that fixes the problem for now. In `BasePresenter` I added this function that I use instead of the translator one:

```php
public function translate($key, $args = []): string
{
	$this->translator->setLocale($this->locale); // This is the important part that fixed the problem
	return $this->translator->translate($key, $args);
}
```

It looks like the `$locale` were somehow being changed to `cs` every time, so I added the `$this->translator->setLocale($this->locale);` line and it actually fixed the issue and sets the proper locale instead of always `cs`.

**Comment by aleswita** [comment]
I have no idea, how can I help you, if you can't revert composer update.

**Comment by matronator** [comment]
Doesn't matter anyway, I think. Even with the solution above there were still issues of some texts still being in Czech, so I did a bit more digging. I found I had the wrong values in `config.neon` for available locale and it was using `cs_CZ` as the default fallback. I had `['en', 'cs', 'ru']` in the config, but it should've been `['en_US', 'ru_RU', 'cs_CZ']` as per the lang file names. Also I set the fallback in `config.neon` to `en_US` so it would override the default `cs_CZ` one (still don't know why Czech in particular was chosen, maybe alphabetical order?).

So it seems like the previous versions before the `composer update` were just less strict about the values and managed to recognize the shortened locale names and assign it to the correct lang files.

So I guess the issue was on my part and this issue can be closed.
